src/components/dialogs/line-type-catalog/line-type-catalog-selector-dialog.js

```diff
@@ -0,0 +1,275 @@
+                    id: 'lineType.circuitsNumber',
+                }),
+                field: 'circuitsNumber',
+                cellRenderer: DefaultCellRenderer,
+                numeric: true,
+            },
+            {
+                headerName: intl.formatMessage({
+                    id: 'lineType.groundWiresNumber',
+                }),
+                field: 'groundWiresNumber',
+                cellRenderer: DefaultCellRenderer,
+                numeric: true,
+            },
+            {
+                headerName: intl.formatMessage({
+                    id: 'lineType.linearResistance',
+                }),
+                field: 'linearResistance',
+                cellRenderer: DefaultCellRenderer,
+                numeric: true,
+            },
+            {
+                headerName: intl.formatMessage({
+                    id: 'lineType.linearReactance',
+                }),
+                field: 'linearReactance',
+                cellRenderer: DefaultCellRenderer,
+                numeric: true,
+            },
+            {
+                headerName: intl.formatMessage({
+                    id: 'lineType.linearCapacity',
+                }),
+                field: 'linearCapacity',
+                cellRenderer: DefaultCellRenderer,
+                numeric: true,
+            },
+        ];
+    }, [intl]);
+
+    // Tries to find in the current tab the preselected row to highlight it.
+    // The props.preselectedRow should be an object like this :
+    // { kind:<string>, type:<string> }
+    // The match is done with its "kind" parameter which should correspond
+    // to the selected tab, and its "type" parameter which should correspond
+    // to the selected line.
+    const highlightSelectedRow = useCallback(() => {
+        const rowToHightlight = selectedRow ?? props?.preselectedRow;
+        if (rowToHightlight && props?.rowData) {
+            const currentTabKind =
+                tabIndex === LineTypeCatalogSelectorDialogTabs.AERIAL_TAB
+                    ? 'AERIAL'
+                    : 'UNDERGROUND';
+            if (rowToHightlight.kind === currentTabKind) {
+                gridRef.current?.api?.forEachNode(function (node) {
+                    node.setSelected(
+                        node.data?.kind === currentTabKind &&
+                            node.data?.type === rowToHightlight.type
+                    );
+                });
+            }
+        }
+    }, [selectedRow, tabIndex, props.preselectedRow, props.rowData]);
+
+    // Tries to highlights the preselected row when changing tabs
+    useEffect(() => {
+        highlightSelectedRow();
+    }, [
+        highlightSelectedRow,
+        selectedRow,
+        tabIndex,
+        props.preselectedRow,
+        props.rowData,
+    ]);
+
+    const headerAndTabs = (
+        <Box
+            sx={{
+                display: 'flex',
+                flexWrap: 'wrap',
+                gap: '15px',
```

Comment:
That's some code lifted from other places. Because 15 is not cleanly divisible by 4, I'd rather left this like this and we could clean all of these instances in another PR.